python/cudf/cudf/core/buffer.py

```diff
@@ -1,218 +1,83 @@
-import pickle
-import types
+import functools
+import operator
 
 import numpy as np
 
 import rmm
+from rmm import DeviceBuffer
 
-from cudf.utils import cudautils, utils
 
-
-class Buffer(object):
-    """A 1D gpu buffer.
-    """
-
-    _cached_ipch = None
-
-    @classmethod
-    def from_empty(cls, mem, size=0):
-        """From empty device array
-        """
-        return cls(mem, size=size, capacity=mem.size)
-
-    @classmethod
-    def null(cls, dtype):
-        """Create a "null" buffer with a zero-sized device array.
+class Buffer:
+    def __init__(self, data=None, size=None, owner=None):
         """
-        mem = rmm.device_array(0, dtype=dtype)
-        return cls(mem, size=0, capacity=0)
-
-    def __init__(self, mem, size=None, capacity=None, categorical=False):
-        if size is None:
-            if categorical:
-                size = len(mem)
-            elif hasattr(mem, "__len__"):
-                if hasattr(mem, "ndim") and mem.ndim == 0:
-                    pass
-                elif len(mem) == 0:
-                    size = 0
-            if hasattr(mem, "size"):
-                size = mem.size
-        if capacity is None:
-            capacity = size
-        self.mem = cudautils.to_device(mem)
-        _BufferSentry(self.mem).ndim(1)
-        self.size = size
-        self.capacity = capacity
-        self.dtype = self.mem.dtype
-
-    def serialize(self):
-        """Called when dask.distributed is performing a serialization on this
-        object.
-
-        Do not use this directly.  It is invoked by dask.distributed.
+        A Buffer represents a device memory allocation.
 
         Parameters
         ----------
-        serialize : callable
-             Used to serialize data that needs serialization .
-        context : dict; optional
-            If not ``None``, it contains information about the destination.
-
-        Returns
-        -------
-        (header, frames)
-            See custom serialization documentation in dask.distributed.
+        data : array_like, int
+            An array-like object or integer representing a
+            device pointer to pre-allocated memory.
```

Comment:
This isn't just device memory:
```suggestion
           pointer to pre-allocated memory.
```

Reply:
Fixed